src/server/utils/utils.py

```diff
@@ -0,0 +1,145 @@
+                    if index % 2 == 0:
+                        demands[int(arr[i+index+1])] = int(arr[i+index+2])
+            elif value == 'STATIONS_COORD_SECTION':
+                for index in range(parameters['STATIONS']):
+                    chargers.add(int(arr[i+index+1]))
+            else:
+                # Building the parameters dictionnary
+                if not value.isdigit() and value.isupper() and value != 'EOF' \
+                        and arr[i+1].replace('.', '', 1).isdigit():
+                    val = ast.literal_eval(arr[i+1])
+                    parameters[value] = val if isinstance(val, float) else int(val)
+
+    # Detect if the file is incomplete
+    if not (parameters or nodes or chargers or demands):
+        # The dataset is missing some parameters
+        raise ValueError('The dataset file is imcomplete')
+
+    return {'VEHICLES': parameters['VEHICLES'], 'CAPACITY': parameters['CAPACITY'],
+            'ENERGY_CAPACITY': parameters['ENERGY_CAPACITY'],
+            'ENERGY_CONSUMPTION': parameters['ENERGY_CONSUMPTION'],
+            'NODES': nodes, 'DEMANDS': demands,
+            'STATIONS': chargers, 'TIME_WINDOWS': time_windows,
+            'DEPOT': parameters['DEPOT_SECTION']}
+
+
+def create_ecvrp(parameters: dict[str, any]) -> ECVRPInstance:
+    """ Create an ECVRPInstance object from the parsed data.
+
+    :param parameters: A parsed dataset object given by parse_dataset()
+    :type parameters: dict[str, any]
+    :return: An ECVRP instance
+    :rtype: ECVRPInstance
+    """
+
+    distance_matrix = compute_distance_matrix(parameters['NODES'])
+
+    # Instantiating the ECVRP instance
+    ecvrp = ECVRPInstance(distance_matrix=distance_matrix, depot_id=parameters['DEPOT'],
+                          chargers=parameters['STATIONS'], demands=parameters['DEMANDS'],
+                          batterie_cost_factor=parameters['ENERGY_CONSUMPTION'],
+                          batterie_charge_rate=1.0, ev_count=parameters['VEHICLES'],
+                          ev_capacity=parameters['CAPACITY'],
+                          ev_battery=parameters['ENERGY_CAPACITY'],
+                          time_windows=parameters['TIME_WINDOWS'])
+
+    return ecvrp
+
+
+def get_datasets(dir_path: str = PATH_TO_DATASETS) -> list[str]:
+    """ List all the files in the dataset folder.
+    """
+    return os.listdir(dir_path)
+
+
+def compute_distance_matrix(nodes: dict[int, tuple[int, int]]) -> list[list[float]]:
+    """ Compute the distance matrix of our nodes
+    """
+    distance_matrix = np.zeros(shape=(len(nodes), len(nodes))).tolist()
+
+    for node, (current_x, current_y) in nodes.items():
+        for next_node, (next_x, next_y) in nodes.items():
+            if node != next_node:
+                dist = np.sqrt(np.power(current_x-next_x, 2) + np.power(current_y-next_y, 2))
+                distance_matrix[node-1][next_node-1] = dist
```

Comment:
I think that either the nodes have to be offsetted when reading or the offset should be parametrized. This issue has already been discussed, but nothing was decided about what to do.